From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anuken <arnukren@gmail.com>
Date: Thu, 28 Sep 2023 20:52:08 +0800
Subject: [PATCH] PICK @client

(cherry picked from commit 4ed6ed9bcbd22f35d0c43f881ffe8f60e1de1fe2)
---
 core/src/mindustry/logic/GlobalVars.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/src/mindustry/logic/GlobalVars.java b/core/src/mindustry/logic/GlobalVars.java
index 57b30bcd7bab28592d4970a5fb1fac7f3e41a58a..a4e7d6a79a2de35715a3e0ed2dbc7da14309127d 100644
--- a/core/src/mindustry/logic/GlobalVars.java
+++ b/core/src/mindustry/logic/GlobalVars.java
@@ -27,7 +27,7 @@ public class GlobalVars{
     public static final Rand rand = new Rand();
 
     //non-constants that depend on state
-    private static int varTime, varTick, varSecond, varMinute, varWave, varWaveTime, varServer;
+    private static int varTime, varTick, varSecond, varMinute, varWave, varWaveTime, varServer, varClient;
 
     private ObjectIntMap<String> namesToIds = new ObjectIntMap<>();
     private Seq<Var> vars = new Seq<>(Var.class);
@@ -57,6 +57,7 @@ public class GlobalVars{
         varWaveTime = put("@waveTime", 0);
 
         varServer = put("@server", 0);
+        varClient = put("@client", 0);
 
         //special enums
         put("@ctrlProcessor", ctrlProcessor);
@@ -152,6 +153,7 @@ public class GlobalVars{
 
         //network
         vars.items[varServer].numval = (net.server() || !net.active()) ? 1 : 0;
+        vars.items[varClient].numval = net.client() ? 1 : 0;
     }
 
     /** @return a piece of content based on its logic ID. This is not equivalent to content ID. */
